--- dlls/propsys/propsys_main.c.orig	2026-08-23 01:10:23.299883196 +0100
+++ dlls/propsys/propsys_main.c	2026-08-23 01:10:23.305371070 +0100
@@ -852,6 +852,12 @@
     {L"System.Devices.GlyphIcon", &PKEY_Devices_GlyphIcon, VT_LPWSTR},
     {L"System.Devices.DevObjectType", &PKEY_Devices_DevObjectType, VT_UI4},
     {L"System.DeviceInterface.Bluetooth.DeviceAddress", &PKEY_DeviceInterface_Bluetooth_DeviceAddress, VT_LPWSTR},
+    {L"System.DeviceInterface.Hid.UsagePage", &PKEY_DeviceInterface_Hid_UsagePage, VT_UI2},
+    {L"System.DeviceInterface.Hid.UsageId", &PKEY_DeviceInterface_Hid_UsageId, VT_UI2},
+    {L"System.DeviceInterface.Hid.VendorId", &PKEY_DeviceInterface_Hid_VendorId, VT_UI2},
+    {L"System.DeviceInterface.Hid.ProductId", &PKEY_DeviceInterface_Hid_ProductId, VT_UI2},
+    {L"System.DeviceInterface.WinUsb.UsbVendorId", &PKEY_DeviceInterface_WinUsb_UsbVendorId, VT_UI2},
+    {L"System.DeviceInterface.WinUsb.UsbProductId", &PKEY_DeviceInterface_WinUsb_UsbProductId, VT_UI2},
     {L"System.Devices.Aep.AepId", &PKEY_Devices_Aep_AepId, VT_LPWSTR},
     {L"System.Devices.Aep.ProtocolId", &PKEY_Devices_Aep_ProtocolId, VT_CLSID},
     {L"System.Devices.Aep.IsPaired", &PKEY_Devices_Aep_IsPaired, VT_BOOL},
